src/Concurrency.h

```diff
@@ -90,14 +89,15 @@ class queuing_rw_mutex {
         --_reader_count;
 
         if ((_reader_count == 0) && (_writer_count > 0)) {
-            --_writer_count;
             dequeue_one_writer();
         }
     }
     void writer_leave() {
         std::unique_lock<std::mutex> lock(_mtx);
-
         --_writer_count;
+        short rc = _reader_count;
```

Comment:
these ones are still here?

Reply:
ok. i will remove and push. The tests have all completed successfully at this point.